src/commonMain/kotlin/com/charleskorn/kaml/YamlScalarInput.kt

```diff
@@ -42,13 +42,22 @@ internal class YamlScalarInput(val scalar: YamlScalar, yaml: Yaml, context: Seri
             return index
         }
 
-        val choices = (0..enumDescriptor.elementsCount - 1)
+        val choices = (0..<enumDescriptor.elementsCount)
+            .asSequence()
             .map { enumDescriptor.getElementName(it) }
-            .sorted()
-            .joinToString(", ")
+
+        if (configuration.decodeEnumCaseInsensitive) {
+            val lowerChoices = choices.map { it.lowercase() }
+            val lowerContent = scalar.content.lowercase()
+
+            val idx = lowerChoices.indexOf(lowerContent)
+            if (idx != -1) {
+                return idx
+            }
```

Comment:
suggestion: You can simplify this part. Instead of create a lowercase version of each element name you can use `equals(ignoreCase = true)` to find the element you need
```suggestion
            val idx = choices.indexOfFirst { it.equals(scalar.content, ignoreCase = true) }
            if (idx != -1) {
                return idx
            }
```